syntax = "proto3";

package musica_app;

service UsuarioService {
    rpc CriarUsuario (Usuario) returns (Resposta);
    rpc LerUsuario (UsuarioID) returns (Usuario);
    rpc AtualizarUsuario (Usuario) returns (Resposta);
    rpc DeletarUsuario (UsuarioID) returns (Resposta);
    rpc LerUsuarios (Empty) returns (UsuarioList);
}

service MusicaService {
    rpc CriarMusica (Musica) returns (Resposta);
    rpc LerMusica (MusicaID) returns (Musica);
    rpc AtualizarMusica (Musica) returns (Resposta);
    rpc DeletarMusica (MusicaID) returns (Resposta);
    rpc LerMusicas (Empty) returns (MusicaList);
}

message Usuario {
    int32 ID = 1;
    string nome = 2;
    int32 idade = 3;
    repeated int32 playlists = 4;
}

message Musica {
    int32 ID = 1;
    string nome = 2;
    string artista = 3;
}

message UsuarioID {
    int32 ID = 1;
}

message MusicaID {
    int32 ID = 1;
}

message Resposta {
    string message = 1;
}

message Empty {}

message UsuarioList {
    repeated Usuario usuarios = 1;
}

message MusicaList {
    repeated Musica musicas = 1;
}
